chore: update Shield copywriting and error msgs cp-13.10.0

lwin-kyaw opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

This PR updates 
- Shield confirmation msgs, copywriting
- add marketing deep link for shield
- Fixes confirmation release blocker introduced by side panel changes [here](https://github.com/MetaMask/metamask-extension/pull/37778)
- Enables shield flag in all builds
- Fixes paused state subscription release blocker 


[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/37829?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry: update Shield confirmation msgs

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->
<img width="489" height="949" alt="Screenshot 2025-11-14 at 2 10 26 PM" src="https://github.com/user-attachments/assets/14db38bc-5f9f-440e-8ff7-c5b9533764f1" />

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Updates Shield copy and paused-state UX, refactors metrics with common props and new restart tracking, adds deep link to show entry modal, classifies approve tx, and enables the Shield feature flag.
> 
> - **Shield UX copy & i18n**:
>   - Replace “membership” with “plan” across Shield strings; refine coverage/renewal/tooltips; split paused-state messages/actions by payment type; minor text tweaks (e.g., Save 17%).
> - **Metrics**:
>   - Add `getShieldCommonTrackingProps`; update all Shield capture hooks to use it.
>   - Replace `ShieldErrorStateClicked` with `ShieldMembershipErrorStateClicked` and adjust event emissions.
>   - Move subscription restart request tracking from background to UI (`captureShieldSubscriptionRestartRequestEvent`).
> - **Deep links & Settings**:
>   - Add `SHIELD_QUERY_PARAMS.showShieldEntryModal`; when true, route to `SETTINGS_ROUTE` and show entry modal or redirect to Shield tab if subscribed.
>   - Export `SETTINGS_ROUTE` via deep-links.
> - **Toasts & Banners**:
>   - Paused toast now shows type-specific descriptions/actions (card/crypto/unexpected).
>   - Settings Shield banner uses new paused/ending-soon copy and actions; adds "Contact support" flow with user identifiers.
> - **Transactions**:
>   - Classify `TransactionType.shieldSubscriptionApprove` in metrics.
>   - Crypto approval flow now navigates to confirmation after approval creation is detected.
> - **Builds**:
>   - Enable `METAMASK_SHIELD_ENABLED: 'true'` in config.
> - **Misc UI**:
>   - Use `color={TextColor.textAlternative}` for billing label; small refactors/imports.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 8069be41eaf61c07e94a19c0b8121a1500ee9e27. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

lwin-kyaw committed:
chore: update Shield confirmation msgs

lwin-kyaw committed:
chore: revert 'shieldCovered' locale

lwin-kyaw committed:
fix: fixed metrics for Subscription Restart

hieu-w committed:
fix: Shield Plan Select page text update (#37833)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/37833?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry:

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**



https://github.com/user-attachments/assets/34016a2e-3938-4fd1-ae4d-357193434bea


<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Updates Shield plan note text and increases displayed savings from 16%
to 17% across en and en_GB locales.
> 
> - **Localization (i18n)**:
>   - **Transaction Shield plan copy**:
> - Update `shieldPlanCryptoMonthlyNote` message to "Total monthly fees
pre-approved for a year".
>     - Change `shieldPlanSave` from `Save 16%` to `Save 17%`.
> - Affected files: `app/_locales/en/messages.json`,
`app/_locales/en_GB/messages.json`.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
bb797c39ba543505b99ddd925b393b70f44fde00. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

lwin-kyaw committed:
feat: enable shield flag in build.yml

lwin-kyaw committed:
Merge remote-tracking branch 'origin/fix/confirmation-shield-messages' into fix/confirmation-shield-messages

lwin-kyaw committed:
chore: removed 'METAMASK_SHIELD_ENABLED' experimental build flag

hieu-w committed:
fix: update shield plan confirm text (#37835)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/37835?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry:

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Updates Transaction Shield English localizations to use “plan”
terminology and clarifies monthly authorization/billing tooltip in both
`en` and `en_GB`.
> 
> - **Localization (en, en_GB)**
> - Terminology: Replace "membership" with "plan" across Transaction
Shield strings (e.g., `shieldConfirmMembership`,
`shieldTxMembershipActive`, `shieldTxMembershipCancel`,
`shieldTxMembershipRenew`, `shieldTxMembershipId`).
> - Notifications/Status: Update payment failure/paused/cancellation
copy (e.g.,
`pushNotificationShieldSubscriptionPaymentFailedDescriptionShort`,
`shieldCoverageAlertMessagePaused`, `shieldTxMembershipErrorPaused`,
`shieldTxMembershipCancelNotification`).
> - Timing: Change ending text to "Plan ends on $1"
(`shieldCoverageEndingDescription`).
> - Billing Tooltip: Clarify monthly authorization text
(`shieldEstimatedChangesMonthlyTooltipText`: "Authorize $1/month for 12
months ($2 total)").
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
d4012c95cd08cf5ba4ac055b89206e5377078292. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

tuna1207 committed:
fix: subscription-error-message (#37836)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

- Correct subscription paused toast and shield settings error message.
- Handle unexpected error message contact support action.
Figma:
https://www.figma.com/design/HTAO1SrmixV4ppv7qIvLoa/Metamask-Transaction-Shield?node-id=14631-57069&p=f&t=AcoJzS3JF4yuRexs-0

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/37836?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry: correct subscription error message in toast and
settings screen

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Updates Shield paused/payment error UX: new localized messages,
context-aware toast and settings actions for card/crypto/unexpected
errors, and a support contact handler with user context.
> 
> - **UI/UX (Transaction Shield)**:
> - **Paused Toast**: Switches to `shieldPaymentPaused` with dynamic
description/action based on payment method (`card`, `crypto`,
`unexpected`).
>   - **Settings Page** (`transaction-shield.tsx`):
> - Context-aware error banner/tooltips and CTA labels for paused states
(card/crypto/unexpected) and ending-soon renewal.
> - Handles crypto-specific flows (add funds, approval tx) and card
update; opens support on unexpected errors.
> - Minor API tweaks (use `useUnCancelSubscription(current)`; refine
payment method rendering).
> - **Hooks**:
> - Adds `useHandleSubscriptionSupportAction()` to open support with
appended `metamask_version`, profile, MetaMetrics, and Shield customer
IDs.
> - **Localization** (`app/_locales/en`, `en_GB`):
> - Adds new `shieldPaymentPaused*` and detailed
`shieldTxMembershipErrorPaused*` keys; updates renewal/insufficient
funds copy; removes deprecated `shieldPaymentDeclined` text.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
b6f6e8ff0b1976a7417374c343254f7fed21a82d. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

lwin-kyaw committed:
resolved conflicts

lwin-kyaw committed:
Merge remote-tracking branch 'origin/fix/confirmation-shield-messages' into fix/confirmation-shield-messages

tuna1207 committed:
fix: locale verify

tuna1207 committed:
fix: navigate to confirmation page if there are pending approvals and shield transaction is dispatched

tuna1207 committed:
fix: correct crypto approval billing detail text color

tuna1207 commented on ui/pages/settings/transaction-shield-tab/transaction-shield.tsx:
```diff
@@ -448,22 +462,32 @@ const TransactionShield = () => {
+        actionButtonLabel =
+          'shieldTxMembershipErrorPausedCryptoInsufficientFundsAction';
+      } else if (isCardPayment) {
+        descriptionText = 'shieldTxMembershipErrorPausedCard';
```
it is expected per design, handler still handle approval error correctly

tuna1207 committed:
fix: isUnexpectedErrorCryptoPayment logic

tuna1207 commented on ui/pages/settings/transaction-shield-tab/transaction-shield.tsx:
```diff
@@ -412,10 +395,39 @@ const TransactionShield = () => {
+  const isUnexpectedErrorCryptoPayment =
+    currentShieldSubscription &&
+    isCryptoPaymentMethod(currentShieldSubscription.paymentMethod) &&
+    Boolean(currentShieldSubscription.paymentMethod.crypto.error);
```
updated

tuna1207 committed:
fix: translation

tuna1207 commented on app/_locales/en_GB/messages.json:
```diff
@@ -6251,55 +6263,64 @@
     "message": "Your payment was declined. Please update your payment method to continue your coverage."
   },
+  "shieldTxMembershipErrorPausedCryptoInsufficientFunds": {
+    "message": "Plan paused due to insufficient funds. Payment updates may take up to $1hours.",
```
updated

tuna1207 commented on ui/pages/settings/transaction-shield-tab/transaction-shield.tsx:
```diff
@@ -416,6 +399,32 @@ const TransactionShield = () => {
+  const isUnexpectedErrorCryptoPayment =
+    currentShieldSubscription &&
+    isCryptoPaymentMethod(currentShieldSubscription.paymentMethod) &&
+    !currentShieldSubscription.paymentMethod.crypto.error;
```
this logic is correct

tuna1207 commented on ui/pages/settings/transaction-shield-tab/transaction-shield.tsx:
```diff
@@ -416,6 +399,32 @@ const TransactionShield = () => {
+  const isUnexpectedErrorCryptoPayment =
+    currentShieldSubscription &&
+    isCryptoPaymentMethod(currentShieldSubscription.paymentMethod) &&
+    !currentShieldSubscription.paymentMethod.crypto.error;
```
this logic is correct

tuna1207 committed:
fix: make sure isUnexpectedErrorCryptoPayment is from paused subscription

tuna1207 committed:
feat: handle shield deep link open entry modal cp-13.10.0 (#37846)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

Add params handle to shield deep link for
`link.metamask.io/shield?showShieldEntryModal=true` which show shield
entry modal in settings screen if user not subscribed yet

[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/37846?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the
CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing
description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and
accurately)
-->

CHANGELOG entry: add shield deep link `showShieldEntryModal` param
handle

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [x] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds `/shield?showShieldEntryModal=true` deep link that routes to
Settings and either opens the Shield entry modal or navigates to the
Shield page if subscribed.
> 
> - **Deep links**
> - Add `SHIELD_QUERY_PARAMS.showShieldEntryModal` and handle in
`shared/lib/deep-links/routes/shield.ts`.
> - When `showShieldEntryModal=true`, redirect to `SETTINGS_ROUTE` with
original query; otherwise go to `SHIELD_PLAN_ROUTE`.
> - Export `SETTINGS_ROUTE` from
`shared/lib/deep-links/routes/route.ts`.
> - **Settings page**
> - Parse `showShieldEntryModal` from URL in `settings.container.js` and
pass `shouldShowShieldEntryModal` prop.
> - In `settings.component.js` `componentDidMount`, if
`shouldShowShieldEntryModal`:
> - If `hasSubscribedToShield`, navigate to `TRANSACTION_SHIELD_ROUTE`.
>     - Else, show `ShieldEntryModal`.
>   - Add `shouldShowShieldEntryModal` to PropTypes.
> 
> <sup>Written by [Cursor
Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit
519b088ad610cfd5e64a2300d5473e4c06d4dab5. This will update automatically
on new commits. Configure
[here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

chaitanyapotti committed:
Merge branch 'main' into fix/confirmation-shield-messages
